x-pack/plugins/enterprise_search/server/routes/enterprise_search/search_applications.ts

```diff
@@ -64,7 +79,18 @@ export function registerSearchApplicationsRoutes({ log, router }: RouteDependenc
       const engine = (await client.asCurrentUser.searchApplication.get({
         name: request.params.engine_name,
       })) as EnterpriseSearchApplication;
-      const indicesStats = await fetchIndicesStats(client, engine.indices);
+
+      let indices: string[];
+      try {
```

Comment:
suggestion: given that we're using this fetch/try/catch logic for aliases in at least two places now, we should probably extract that to a common function, or embed this logic in `fetchAliasIndices` so it always returns an empty array if an alias is missing.

Reply:
I was thinking of doing this, but the UIs treat the case of missing indices differently which also depends on the API response.
the list API does not return an error, while the `field_capabilities` API returns an error when the alias is missing.
so I thought it was cleaner to catch the error at the API level rather than conditionally check whether `indices` is empty.